api/src/users/users.service.ts

```diff
@@ -192,7 +193,67 @@ export class UsersService {
         return {
           message: 'User updated successfully.',
           statusCode: HttpStatus.OK,
+          data: users,
+        };
+      } else {
+        return {
+          message: 'User not found.',
+          statusCode: HttpStatus.NOT_FOUND,
+          data: {},
+        };
+      }
+    } catch (error) {
+      if (error instanceof HttpException) {
+        return {
+          message: error.message,
+          statusCode: HttpStatus.CONFLICT,
+          data: {},
+        };
+      } else {
+        return {
+          message: error.message,
+          statusCode: HttpStatus.INTERNAL_SERVER_ERROR,
+          data: {},
+        };
+      }
+    }
+  }
 
+  async updatePassword(
+    userDto: UsersDto,
+    user: { id: any; password: string | undefined },
+    id: number,
+    currentPassword: string,
+  ): Promise<{ message: string; statusCode: number; data: object }> {
+    const { password } = userDto;
+    try {
+      const existingUser = await this.userModel.findOne({
+        where: { id: id },
+      });
+      if (existingUser) {
+        const isPasswordMatch = await compare(
+          currentPassword,
+          existingUser.password,
+        );
+        if (!isPasswordMatch) {
+          return {
+            message: 'Current password is incorrect.',
+            statusCode: 422,
+            data: {},
```

Comment:
Remove data object from the response if it's not needed

Reply:
removed data object